lib/screens/home/home_screen.dart

```diff
@@ -62,10 +71,24 @@ class _HomeScreenState extends ConsumerState<HomeScreen> {
               minHeight: MediaQuery.of(context).size.height,
               maxHeight: MediaQuery.of(context).size.height,
             ),
-            child: HomePagingWidget(surveys: surveys),
+            child: HomePagingWidget(
+                surveys: surveys,
+                onPageChanged: (index) {
+                  _currentPageIndex.value = index;
+                }),
           ),
         ),
       ),
     );
   }
+
+  void _setupListener() {
+    _currentPageIndex.addListener(() {
+      final currentPageIndex = _currentPageIndex.value;
+      final surveys = ref.read(_surveysStreamProvider).value ?? [];
+      if (currentPageIndex == surveys.length - 2) {
```

Comment:
We should add a flag to prevent it's called again when we are getting surveys 🤔

Reply:
Updated in [00e97a8](https://github.com/nimblehq/flutter-ic-lydia-ryan/pull/106/commits/00e97a8088230436e9d844d17969de2ad4313205) 🙏🏻